services/speech_to_text/v1.js

```diff
@@ -74,121 +215,102 @@ module.exports = function (RED) {
-          };
-
+          node.error(message_err_format, msg)
+          return;  
         }
+      }  
 
-
-        node.status({fill:'blue', shape:'dot', text:'requesting'});
-        speech_to_text.recognize(params, function (err, res) {
-          node.status({})
+      // We are now ready to process the input data 
+      // If its a buffer then need to read it all before invoking the service 
+      var params = {};
+      if (msg.payload instanceof Buffer) {
+        temp.open({suffix: '.' + fileType(msg.payload).ext}, function (err, info) {
           if (err) {
-            node.error(err, msg);
-          } else {
-            msg.transcription = '';
-            if (res.results.length && res.results[0].alternatives.length) {
-              msg.transcription = res.results[0].alternatives[0].transcript;
-            }
-          }
+            this.status({fill:'red', shape:'ring', text:'unable to open audio stream'});          
+            var message ='Node has been unable to open the audio stream'; 
+            node.error(message, msg);
+            return;        
+          }  
 
-          node.send(msg);
-          if (cb) {
-            cb();
-          }
-        });
-      };
+          stream_buffer(info.path, msg.payload, function (format) {
+            var audio = fs.createReadStream(info.path);
 
-      var stream_buffer = function (file, contents, cb) {
-        fs.writeFile(file, contents, function (err) {
-          if (err) {
-            throw err;
-          }
-          cb(fileType(contents).ext)
-        });
-      };
-
-      var stream_url = function (file, location, cb) {
-        var wstream = fs.createWriteStream(file);
-
-        wstream.on('finish', function () {
-          fs.readFile(file, function (err, buf) {
-            if (err) {
-              throw err;
-            }
-            cb(fileType(buf).ext)
-          })
+            performAction(audio, format, actionComplete, temp.cleanup);        
+          });
         });
+      } else if (urlCheck(msg.payload)) {
+        temp.open({suffix: '.audio'}, function(err, info){
+          if (err) {
+            this.status({fill:'red', shape:'ring', text:'unable to open url audio stream'});          
+            var message ='Node has been unable to open the url audio stream'; 
+            node.error(message, msg);
+            return;        
+          }  
 
-        request(location)
-        .pipe(wstream);
-      };
+          stream_url(info.path, msg.payload, function (format) {
+            var audio = fs.createReadStream(info.path);
 
-      temp.open({suffix: '.audio'}, function (err, info) {
-        if (err) throw err;
+            performAction(audio, format, actionComplete, temp.cleanup);        
+          });
+        });
+      } else {
+        this.status({fill:'red', shape:'ring', text:'payload is invalid'});          
+        var message ='Payload must be either an audio buffer or a string representing a url'; 
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2636061996/issues/source?bid=3225115&fileBranchId=3319654#l301)